src/server/dflycmd.cc

```diff
@@ -202,12 +144,114 @@ void DflyCmd::HandleJournal(CmdArgList args, ConnectionContext* cntx) {
   return rb->SendError(reply, kSyntaxErrType);
 }
 
-uint32_t DflyCmd::AllocateSyncSession() {
+void DflyCmd::Thread(CmdArgList args, ConnectionContext* cntx) {
+  RedisReplyBuilder* rb = static_cast<RedisReplyBuilder*>(cntx->reply_builder());
+  util::ProactorPool* pool = shard_set->pool();
+
+  if (args.size() == 2) {  // DFLY THREAD : returns connection thread index and number of threads.
+    rb->StartArray(2);
+    rb->SendLong(ProactorBase::GetIndex());
+    rb->SendLong(long(pool->size()));
+    return;
+  }
+
+  // DFLY THREAD to_thread : migrates current connection to a different thread.
+  string_view arg = ArgS(args, 2);
+  unsigned num_thread;
+  if (!absl::SimpleAtoi(arg, &num_thread)) {
+    return rb->SendError(kSyntaxErr);
+  }
+
+  if (num_thread < pool->size()) {
+    if (int(num_thread) != ProactorBase::GetIndex()) {
+      listener_->Migrate(cntx->owner(), pool->at(num_thread));
+    }
+
+    return rb->SendOk();
+  }
+
+  rb->SendError(kInvalidIntErr);
+  return;
+}
+
+void DflyCmd::Sync(CmdArgList args, ConnectionContext* cntx) {
+  RedisReplyBuilder* rb = static_cast<RedisReplyBuilder*>(cntx->reply_builder());
+  string_view masterid = ArgS(args, 2);
+  string_view sync_id_str = ArgS(args, 3);
+  string_view flow_id_str = ArgS(args, 4);
+
+  unsigned flow_id;
+  VLOG(1) << "Got DFLY SYNC " << masterid << " " << sync_id_str << " " << flow_id_str;
+
+  if (masterid != sf_->master_id()) {
+    return rb->SendError("Bad master id");
+  }
+
+  if (!absl::SimpleAtoi(flow_id_str, &flow_id) || !absl::StartsWith(sync_id_str, "SYNC")) {
+    return rb->SendError(kSyntaxErr);
+  }
+
+  auto [sync_id, sync_info] = GetSyncInfoOrReply(sync_id_str, rb);
+  if (!sync_id)
+    return;
+
+  // assuming here that shard id and thread id is the same thing.
+  if (int(flow_id) != ProactorBase::GetIndex()) {
+    listener_->Migrate(cntx->owner(), shard_set->pool()->at(flow_id));
+  }
+
+  (void)sync_id;
+  (void)sync_info;
+
+  return rb->SendOk();
+}
+
+void DflyCmd::Expire(CmdArgList args, ConnectionContext* cntx) {
```

Comment:
the same

Reply:
I commented in the header file. I thought that brief comment should belong to headers, whereas implementation details to the source